registry/storage/cache/memory/memory.go

```diff
@@ -156,3 +169,17 @@ func (rsimbdcp *repositoryScopedInMemoryBlobDescriptorCache) SetDescriptor(ctx c
 	rsimbdcp.parent.lru.Add(key, desc)
 	return rsimbdcp.parent.SetDescriptor(ctx, dgst, desc)
 }
+
+// Memory configures inmemory cache
+type Memory struct {
+	Size int `yaml:"size,omitempty"`
+}
+
+// NewCacheOptions returns new memory cache options.
+func NewCacheOptions(size int) map[string]interface{} {
+	return map[string]interface{}{
+		"params": map[interface{}]interface{}{
+			"size": size,
+		},
+	}
+}
```

Comment:
Yes, it's one of a few things in this PR that I'm still mulling over. Does not feel right I agree. 

Note that the Azure storage driver is also decoding auth config in its constructor: thats where I got some inspiration, but I'm not happy about it no.

It's also one of the reasons I marked this PR as draft. There are things to be figured out here.

Reply:
Just FYI: this is a helper func, you don't have to use it, there is nothing that prevents you from providing your own `map[string]interface{}` params in a similar way it's done in the storage drivers.

I was trying to avoid this (and other storage driver config) absolute state of things https://github.com/distribution/distribution/blob/04e3bdaa7c6304d14970f13c01f0a8ded8289e8d/registry/storage/driver/s3-aws/s3.go#L192-L472

But maybe I'm trying to be too clever here 🤷‍♂️

Reply:
Code needs to exist somewhere which transforms a generic `map[string]any` configuration blob into something the implementation can consume. There's no getting away from the code in the body of `s3.FromParameters` aside from leveraging metaprogramming (e.g. `mapstructure`) to do the same thing less verbosely. The only notes I have for `s3.FromParameters` are:
1. It would be easier to unit-test if the code to parse `parameters` was extracted to its own `func(map[string]any) (DriverParamters, error)`
2. `FromParameters` probably does not need to be exported since it is made available through the driver registry. Code which does not need to go through the registry **and therefore does not need to be generic over all storage drivers** does not need to go through the generic parameter-parsing machinery.
3. The functional options pattern would make it easier to evolve the configuration and implementation in semver-compatible ways than to export the `DriverParameters` struct.

Aside from that, I firmly believe that the cache provider implementations _should_ be following the way the s3 storage driver does configuration and construction. Consumers that do not need to be generic over all implementations of an interface should not be forced to use the weakly-typed, generic-over-all-implementations factory interface targeted at humans. Configuration as `map[string]any` makes runtime errors out of what could have otherwise been caught at compile time.